src/widgets/helper/NotebookTab.cpp

```diff
@@ -381,6 +383,65 @@ void NotebookTab::setHighlightState(HighlightState newHighlightStyle)
     this->update();
 }
 
+void NotebookTab::setHighlightState(HighlightState newHighlightStyle,
+                                    ChannelView &channelViewSource,
+                                    MessagePtr message)
+{
+    if (this->isSelected())
+    {
+        return;
+    }
+
+    if (!this->highlightEnabled_ &&
+        newHighlightStyle == HighlightState::NewMessage)
+    {
+        return;
+    }
+
+    if (this->highlightState_ == newHighlightStyle ||
+        this->highlightState_ == HighlightState::Highlighted)
+    {
+        return;
+    }
+
+    auto splitContainer =
+        dynamic_cast<SplitContainer *>(this->notebook_->getSelectedPage());
+    if (splitContainer != nullptr)
+    {
+        const auto &splits = splitContainer->getSplits();
+        for (const auto &split : splits)
+        {
```

Comment:
setHighlightState is called from each "source" split, i.e. if message is received on channel X, then each split with channel X calls this function, and we always only check currently being shown tab and compare it to the source split

therefore "splits" and "split" here refers to currently being shown/selected tab
and source refers to the split with a channel which received a message in another tab